aidial_interceptors_sdk/chat_completion/adapter.py

```diff
@@ -1,45 +1,61 @@
 import json
 import logging
-from typing import Any, AsyncIterator, Type, cast
+from typing import AsyncIterator, Type, cast
 
 from aidial_sdk.chat_completion import ChatCompletion as DialChatCompletion
 from aidial_sdk.chat_completion import Request as DialRequest
 from aidial_sdk.chat_completion import Response as DialResponse
+from aidial_sdk.chat_completion.chunks import DefaultChunk
+from aidial_sdk.exceptions import HTTPException as DialException
 from openai import AsyncStream
 from openai.types.chat.chat_completion import ChatCompletion
 from openai.types.chat.chat_completion_chunk import ChatCompletionChunk
 
+from aidial_interceptors_sdk.chat_completion.annotated_value import (
+    AnnotatedException,
+    Annotation,
+)
 from aidial_interceptors_sdk.chat_completion.base import (
     ChatCompletionInterceptor,
+    RequestDict,
 )
 from aidial_interceptors_sdk.dial_client import DialClient
 from aidial_interceptors_sdk.error import EarlyStreamExit
 from aidial_interceptors_sdk.utils._debug import debug_logging
 from aidial_interceptors_sdk.utils._exceptions import dial_exception_decorator
+from aidial_interceptors_sdk.utils._http_client import HTTPClientFactory
 from aidial_interceptors_sdk.utils._reflection import call_with_extra_body
 from aidial_interceptors_sdk.utils.streaming import (
     block_response_to_streaming_chunk,
-    handle_streaming_errors,
     map_stream,
+    materialize_streaming_errors,
     singleton_stream,
 )
 
 _log = logging.getLogger(__name__)
-_debug = _log.isEnabledFor(logging.DEBUG)
+
+
+def _debug():
```

Comment:
The log level **is being changed** in `configure_loggers`.

If we leave `_debug` constant as is, it will be computed during the module import and it won't pick up the changes in the log level.

I've **inlined** the definition of the `_debug()` method. It's more typical pattern of debug checks: 

https://github.com/openai/openai-python/blob/6974a981aec1814b5abba429a8ea21be9ac58538/src/openai/_base_client.py#L452-L453